src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,250 @@
+                                                                                    homeFireStation, maxNumCrew,
+                                                                                    curNumCrew, maxWaterVolumeLitres,
+                                                                                    curWaterVolumeLitres,
+                                                                                    VehicleStatus::Available,
+                                                                                    workFactor)));
+                std::cout << "Adding vehicle #" << vehicleId << " to global list" <<std::endl;
+            }
+            else if (vehicleType == "Ladder")
+            {
+                districtVehicles.push_back( std::make_unique<FireLadder>(FireLadder(vehicleId, location,
+                                                                                    homeFireStation, maxNumCrew,
+                                                                                    curNumCrew, maxWaterVolumeLitres,
+                                                                                    curWaterVolumeLitres,
+                                                                                    VehicleStatus::Available,
+                                                                                    workFactor)));
+                std::cout << "Adding vehicle #" << vehicleId << " to global list" <<std::endl;
+            }
+        }
+    }
+
+    void DistrictResources::processFireStationSampleData(const std::string &fileName)
+    {
+        std::vector<std::vector<std::string>> stations;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            stations.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : stations)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+
+            auto const x = stof(entry.at(1));
+            auto const y = stof(entry.at(2));
+            auto const location = Location(x, y);
+            auto const stationId = stoi(entry.at(0));
+            auto const maxCrew = stoi(entry.at(3));
+            auto const curCrew = stoi(entry.at(4));
+
+            districtFireStations.push_back( std::make_unique<FireStation>( FireStation(stationId, location, maxCrew, curCrew) ));
+
+            std::cout << "Adding Fire Station #" << stationId << " to global list" << std::endl;
+        }
+    }
+
+    void DistrictResources::processEventSampleData(const std::string &fileName, std::vector<Event> *pendingQueue)
```

Comment:
```suggestion
    void DistrictResources::processEventSampleData(const std::string &fileName, std::vector<Event> & pendingQueue)
```
Is it possible to pass `pendingQueue` by reference here? (References are safer than pointers when they can be used because they cannot be null.)